Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/FareCalculator.hs

```diff
@@ -178,10 +178,14 @@ calculateFareParameters ::
   m FareParameters
 calculateFareParameters params = do
   logTagInfo "FareCalculator" $ "Initiating fare calculation for organization " +|| params.farePolicy.merchantId ||+ " and vehicle service tier " +|| params.farePolicy.vehicleServiceTier ||+ ""
+  now <- getCurrentTime
   let fp = params.farePolicy
+      rideEndTime = case (params.actualRideDuration, params.estimatedRideDuration) of
+        (Just duration, _) -> addUTCTime (secondsToNominalDiffTime duration) params.rideTime
+        (_, Just duration) -> addUTCTime (secondsToNominalDiffTime duration) params.rideTime
+        _ -> now
```

Comment:
Trip end can't be now, right?

Reply:
This is kind of an impossible case, when we don't have both estimated and actual duration. Fare calculation happens at estimate stage(their we pass estimated) and during ride end(their we pass actual - or now -both means the same)